scripts/Reddit-Image-Scrapper/main.py

```diff
@@ -0,0 +1,153 @@
+import json
+import requests
+import urllib.request
+from os import path
+
+dict = {
+    1: {
+        1: "pics",
+        2: "PhotoshopBattles",
+        3: "perfecttiming",
+        4: "itookapicture",
+        5: "Pareidolia",
+        6: "ExpectationVSReality",
+        7: "dogpictures",
+        8: "misleadingthumbnails",
+        9: "FifthWorldPics",
+        10: "TheWayWeWere",
+        11: "pic",
+        12: "nocontextpics",
+        13: "miniworlds",
+        14: "foundpaper",
+        15: "images",
+        16: "screenshots",
+    },
+    2: {
+        1: "mildlyinteresting",
+        2: "interestingasfuck",
+        3: "damnthatsinteresting",
+        4: "beamazed",
+        5: "reallifeshinies",
+        6: "thatsinsane",
+        7: "playitagainsam",
+    },
+    3: {
+        1: "gentlemanboners",
+        2: "prettygirls",
+        3: "hardbodies",
+        4: "girlsmirin",
+        5: "thinspo",
+        6: "goddesses",
+        7: "shorthairedhotties",
+        8: "fitandnatural",
+        9: "wrestlewiththeplot",
+        10: "skinnywithabs",
+        11: "bois",
+        12: "GentlemanBonersGifs",
+    },
+    4: {
+        1: "asiancuties",
+        2: "asiangirlsbeingcute",
+    },
+    5: {
+        1: "PhotoshopBattles",
+        2: "ColorizedHistory",
+        3: "reallifedoodles",
+        4: "HybridAnimals",
+        5: "colorization",
+    },
+    6: {
+        1: "amiugly",
+        2: "roastme",
+        3: "rateme",
+        4: "uglyduckling",
+        5: "prettygirlsuglyfaces",
+    },
+    7: {
+        1: "wallpapers",
+        2: "wallpaper",
+        3: "Offensive_Wallpapers",
+    },
+}
```

Comment:
Instead of hard coding all the possible subreddits, why not make a request to the subreddit provided by the user. Based on the response we can treat it as a valid subreddit or ask the user to re-enter the name of a valid subreddit. 

**Note:** When we follow this approach we also have to make sure that the subreddit has images, if not the .json response won't be able to find any URLs, so we need a check for that as well.

Reply:
Yea for that , I have given 2 options :
* User can enter SubReddit by his/her own
* or, they can use these pre suggested SubReddit under Image Category . which I have collected from [here](https://www.reddit.com/r/ListOfSubreddits/wiki/listofsubreddits/#wiki_images)

and in json if key : `url_overridden_by_dest` doesn't contains any url ending with `.jpg` or `.png` then the loop will `pass`.

Reply:
Can you do something about the women, asians and the ugly categories? First two are a bit nsfw and the last one might be offensive to some people. Also remove the Offensive_wallpapers under wallpaper category if you can.

Reply:
Sure